app/models/embed/purl.rb

```diff
@@ -266,8 +287,10 @@ def image?
           mimetype =~ %r{image/jp2}i
         end
 
+        # Return the size of the file in bytes as an integer or nil if it is not present.
         def size
-          @file.attributes['size'].try(:value)
+          bytes = @file.attributes['size'].try(:value)
+          bytes.blank? ? nil : bytes.to_i
```

Comment:
```suggestion
          bytes.to_i
```

both `nil.to_i` and `''.to_i` both return 0

Reply:
Why is that important?  Where do we use that distinction? It looks to me like we have to go out of our way and add complexity to the code to accomodate this distinction without any user benefit.

Reply:
I changed size to return an integer rather than a string. There was a test which broke when I started returning 0 instead of nil. Would it be helpful to know which test that was?

Reply:
Yes.

Reply:
These three tests failed when I simply changed `ResourceFile.size` to return an integer:

```
rspec ./spec/models/embed/purl_spec.rb:268 # Embed::Purl Purl::Resource Purl::Resource::ResourceFile attributes gets the size from the size attribute
rspec ./spec/views/embed/template/_file.html.erb_spec.rb:77 # embed/template/_file without file size displays Download as the link text
rspec ./spec/views/embed/template/_file.html.erb_spec.rb:88 # embed/template/_file with empty file size displays Download as the link text
```

Reply:
Let's do this then. https://github.com/sul-dlss/sul-embed/pull/1426 I think it's more important to us to end this work with simple, maintainable code then it is to value changing the fewest number of lines.